x-pack/plugins/synthetics/server/synthetics_service/service_api_client.ts

```diff
@@ -165,9 +165,8 @@ export class ServiceAPIClient {
     const promises: Array<Observable<unknown>> = [];
 
     this.locations.forEach(({ id, url }) => {
-      const locMonitors = allMonitors.filter(
-        ({ locations }) =>
-          !locations || locations.length === 0 || locations?.find((loc) => loc.id === id)
+      const locMonitors = allMonitors.filter(({ locations }) =>
+        locations?.find((loc) => loc.id === id && loc.isServiceManaged)
```

Comment:
We do this a lot, might want to create helpers for filtering for private locations. Though, that would add one more import line and the logic is tiny. Up to you.